plugins/offline-storage.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Offline Storage plugin
+ * 
+ * Stores errors failed to get send and try to send them when
+ * Networkf comes back or on init
+ */
+
+var offlineStorageKey = 'raven-js-offline-queue';
+
+function offlineStoragelugin(Raven, options) {
+  options = options || {};
+
+  function processOfflineQueue() {
+    // Let's stop here if there's no connection
+    if (!navigator.onLine) {
+      return;
+    }
+
+    try {
+      // Get the queue
+      var queue = JSON.parse(localStorage.getItem(offlineStorageKey)) || [];
+
+      // Store an empty queue. If processing these one fails they get back to the queue
+      localStorage.setItem(offlineStorageKey, JSON.stringify([]));
```

Comment:
We can call `removeItem` instead to be more direct. It'll fallback to `[]` anyway during processing

Reply:
true :)